specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2023-03-15-preview/managedCassandra.json

```diff
@@ -0,0 +1,2631 @@
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/CassandraClusterRepairPublicResource"
+            },
+            "description": "The properties specifying the desired state of repairRun to be created."
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Accepted. Repair run will be created shortly.",
+            "headers": {
+              "azure-AsyncOperation": {
+                "type": "string",
+                "description": "URI to poll for completion status."
+              },
+              "location": {
+                "type": "string",
+                "description": "URI to poll for completion status."
+              }
+            }
+          },
+          "200": {
+            "description": "Success. Repair is created successfully.",
+            "schema": {
+              "$ref": "#/definitions/CassandraReaperRunStatus"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/cassandraClusters/{clusterName}/repairRuns/{repairRunId}/owner/{ownerName}": {
+      "delete": {
+        "operationId": "CassandraRepair_Delete",
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "CosmosDBManagedCassandraClusterRepairRunDelete": {
+            "$ref": "./examples/CosmosDBManagedCassandraClusterRepairRunDelete.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/clusterNameParameter"
+          },
+          {
+            "$ref": "#/parameters/repairRunId"
+          },
+          {
+            "$ref": "#/parameters/ownerName"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "202": {
+            "headers": {
+              "Location": {
+                "type": "string"
+              }
+            },
+            "description": "Success. The async operation to delete repair run will begin as soon as possible."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "cosmos-db.json#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/cassandraClusters/{clusterName}/repairRuns/list": {
```

Comment:
>/list" [](http://example.com/codeflow?start=145&length=6)

This is not following the ARM API standards. list operation here cannot be a post. You are performing a collection GET on Repair types technically. Can you model Repair as a Resource type?

Reply:
The reason it cannot be Get is because, this is an async operation, the status is returned as header response and after polling the list of repairRuns are returned.

Reply:
This is not a valid URI segment for a POST action name.  POST actions are only valid on a resource or on a resource provider.  Here its a POST action on a "repairRuns" "resourceType".  [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests)

Reply:
Hi @pilor can you please provide an exception in this case, our Control Plane changes are already deployed. This being a preview swagger I can open a bug and fix it the next release. That would be greatly appreciated without risking the feature. Thank you.

Reply:
This is the public swagger repo.  When this gets released it will be a public API version and the changes you are talking about would be breaking changes.  You can go through the breaking change review in advance (see the PR description for links on how to do that) to see if the changes would be allowed in a future API version if you wish.

Reply:
Hi all, we have removed the changes in this file. Please re-review. Thanks.